### PR TITLE
INTG-845 Refactor PaymentSummary into re-usable component and impleme…

### PR DESCRIPTION
**Pull request by jwsandeman**
…nt in OrderFinished, TransactionDetails and SplitPage components

### CODE REVIEW

File: src/components/OrderFinished/OrderFinished.tsx [hunk 1]
```diff
@@ -1,53 +1,41 @@
...
+import { TxLogItem, TxLogService } from '../../services/txLogService';
 
 function OrderFinished(): React.ReactElement {
-  const classes = useStyles();
   const selectedTerminal = useSelector(selectedTerminalIdSelector);
   const currentTerminal = useSelector(terminalInstance(selectedTerminal)) as ITerminalProps;
   const { typePath } = useSelector(terminalTransactionTypeObject(selectedTerminal));
-  const isTxFinished = useSelector(terminalTxFlowFinishedTracker(selectedTerminal));
-  const isTxSuccess = useSelector(isTerminalTxFlowSuccess(selectedTerminal));
+  const [currentTransaction, setcurrentTransaction] = useState<TxLogItem>();
```
Comment (ruoxin-mx51) [inline src/components/OrderFinished/OrderFinished.tsx:20]: Could the setter name be changed to `setCurrentTransaction`?

File: src/components/PaymentSummary/OrderButtons.tsx
Comment (ruoxin-mx51) [inline src/components/PaymentSummary/OrderButtons.tsx:1]: The OrderButton has taken on too many responsibilities that are not its own. I recommend changing to take 2 parameters: the button name and the callback function when the button is clicked. Or just take the whole button as a parameter.

File: src/components/PaymentSummary/PaymentSummary.tsx [hunk 2]
```diff
@@ -1,173 +1,93 @@
...
+            <Box className={classes.root}>
+              <Box flexGrow="2" className={classes.roots}>
+                <OrderStatus currentTransaction={currentTransaction} />
 
-  const transactionStatus = useMemo(() => {
-    const transactionType = getTxTypeByPosRefId(txFlow?.posRefId ?? '').toUpperCase();
-    const status = txFlow?.success === 'Success' ? 'APPROVED' : 'DECLINED';
-    return `${transactionType} ${status}`;
-  }, [txFlow]);
+                {splitTransaction ? (
```
Comment (ruoxin-mx51) [inline src/components/PaymentSummary/PaymentSummary.tsx:66]: Similar to OrderButton, I suggest adding a "custom OrderInfo" parameter and pass the SplitSummary as a custom OrderInfo from SplitReceipt page. With this change the PaymentSummary component will no longer need to know the details of split transaction.

**Review by ruoxin-mx51 — CHANGES_REQUESTED** [review]
The new receipt looks great. Just some suggestions about the structure.

**Review by rossi-mx51 — APPROVED** [review]

**Pull request closed without merging**